packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/CodeModelPlugin.cs

```diff
@@ -64,17 +65,22 @@ protected CodeModelPlugin()
         public virtual OutputLibrary OutputLibrary { get; } = new();
         public virtual InputLibrary InputLibrary => _inputLibrary.Value;
         public virtual TypeProviderWriter GetWriter(TypeProvider provider) => new(provider);
-        public virtual IReadOnlyList<MetadataReference> AdditionalMetadataReferences => [];
+        public IReadOnlyList<MetadataReference> AdditionalMetadataReferences => _additionalMetadataReferences;
 
         public virtual void Configure()
         {
         }
 
-        public virtual void AddVisitor(LibraryVisitor visitor)
+        public void AddVisitor(LibraryVisitor visitor)
         {
             _visitors.Add(visitor);
         }
 
+        public void AddMetadataReferences(params MetadataReference[] references)
+        {
+            _additionalMetadataReferences.AddRange(references);
+        }
```

Comment:
since in visitors we use `AddVisitor` which only adds one
then should we apply the same pattern here?

Reply:
yeah lets make them consistent.